swabbie-redis/src/main/kotlin/com/netflix/spinnaker/swabbie/redis/RedisWorkQueue.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.swabbie.redis
+
+import com.netflix.spinnaker.kork.jedis.RedisClientDelegate
+import com.netflix.spinnaker.kork.jedis.RedisClientSelector
+import com.netflix.spinnaker.swabbie.model.WorkConfiguration
+import com.netflix.spinnaker.swabbie.model.WorkItem
+import com.netflix.spinnaker.swabbie.work.WorkQueue
+import org.springframework.stereotype.Component
+
+/**
+ * Stores an list of [com.netflix.spinnaker.swabbie.model.WorkItem]
+ * @property _seed work configurations defined in the application config from which to derive work items
+ * @see [com.netflix.spinnaker.swabbie.work.WorkQueueManager]
+ */
+@Component
+class RedisWorkQueue(
```

Comment:
The current implementation will result in the queue containing N duplicate workItems when running N swabbie instances, since each will call `seed()` without restriction on what gets pushed onto the queue. Using a redis set that is added to via `sadd` and pulled from via `spop` would solve this. Should also remove the need to shuffle the workItems, since `spop` read order should be pseudo-random.